client.go

```diff
@@ -100,29 +97,24 @@ func (c *Client) Consumer(topics []string, cfg rdkafka.ConfigMap) (*Consumer, er
 	// topic
 	cid := ConsumerID(fmt.Sprintf("%s|%s", c.id, strings.Join(topics, ",")))
 
-	for _, topic := range topics {
-		if existingID, ok := c.consByTopic[topic]; ok {
-			if existingID != cid {
-				return nil, fmt.Errorf("Topic %s has another consumer", topic)
-			}
-		}
+	if c.consID == "" {
+		c.consID = cid
 	}
 
-	c.consumers[cid] = true
-	for _, topic := range topics {
-		c.consByTopic[topic] = cid
+	if cid != c.consID {
+		return nil, fmt.Errorf("Client '%s' has already Consumer '%s' registered.", c.id, c.consID)
 	}
 
 	return c.consManager.GetOrCreate(cid, c.consGID, topics, cfg)
 }
 
-func (c *Client) ConsumerByTopic(topic string) (*Consumer, error) {
-	consumerID, ok := c.consByTopic[topic]
-	if !ok {
-		return nil, fmt.Errorf("No consumer for topic %s", topic)
+// fetchConsumer returns the registered Consumer for the current Client.
+func (c *Client) fetchConsumer() (*Consumer, error) {
+	if c.consID == "" {
+		return nil, errors.New("No consumer is registered for Client " + c.id)
```

Comment:
I don't think we should return an error in this case. It means a consumer does not exist, so I wouldn't consider this an exceptional case.

Reply:
ditto! in the updated code path we don't raise an error in that case:

```
if c.consumer == nil {
        cons, err := c.registerConsumer(cid, c.consGID, topics, cfg)
	if err != nil {
		return nil, fmt.Errorf("Failed to register Consumer '%s' for Client '%s'", cid, c.id)
	}
	c.consumer = cons
} else if cid != c.consumer.id {
	return nil, fmt.Errorf("Client '%s' has already Consumer '%s' registered.", c.id, c.consumer.id)
}
```